### PR TITLE
[mds-*] Append commit hash to image tag if on non-master branch

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

File: bin/mdsctl [hunk 1]
```diff
@@ -311,13 +311,17 @@ installImages() {
...
-  docker build -t ${1}$(getBranchSuffix) .
+  docker build -t ${1}$(getBranchSuffix)$(getCommitHash) .
 }
 
 getBranchSuffix() {
   git rev-parse --abbrev-ref HEAD | sed -E -e 's/[^-a-zA-Z0-9]/-/g' -e 's/^/-/'
 }
 
+getCommitHash() {
+  git rev-parse HEAD | sed -E -e 's/[^-a-zA-Z0-9]/-/g' -e 's/^/-/'
```
Comment (macsj200) [inline bin/mdsctl]: When I run this command I get
`-366bf96cc1bad92cc0b79c4a7951dde536208513`
I guess we want that `-` character at the beginning? Also, that's the full hash, not the shortened one that's normally shown with commits (which is probably fine).

Comment (drtyh2o) [inline bin/mdsctl]: `git rev-parse --short HEAD`

File: bin/mdsctl [hunk 2]
```diff
@@ -333,7 +337,13 @@ templateDockerCompose() {
 
 # TODO: DRY
 templateValues() {
-  suffix=$(getBranchSuffix)
+  branchSuffix=$(getBranchSuffix)
+  if [ "$branchSuffix" = "-master" ]; then
```
Comment (drtyh2o) [inline bin/mdsctl]: Wouldn't including the commit hash make sense for `master` as well or are we assuming the version number will be sufficiently descriptive?

File: bin/mdsctl (hunk 1 above)
Comment (avatarneil) [reply bin/mdsctl]: Done 👍

Comment (avatarneil) [reply bin/mdsctl]: Yeup, that's intended indeed. Just serves as a delimiter in the image tag.

File: bin/mdsctl (hunk 2 above)
Comment (avatarneil) [reply bin/mdsctl]: @invertigo Can probably weigh in more, but for anything in `master` the version number should always match up with one commit. This isn't necessarily the case on other branches (as we make changes without bumping version on each merge to develop)

File: bin/mdsctl [hunk 3]
```diff
@@ -333,7 +337,13 @@ templateDockerCompose() {
...
+    suffix="$branchSuffix"
+  else
+    suffix="$branchSuffix"$(getCommitHash)
```
Comment (drtyh2o) [inline bin/mdsctl]: You could insert the `-` here instead of using `sed` in `getCommitHash` since the `-` isn't actually part of the hash.

Comment (avatarneil) [reply bin/mdsctl]: Yeah, I was just following the pattern from `getBranchSuffix()`. I'm happy with either!

**Pull request merged**